### PR TITLE
[AST-43] Flag mapped pins that have not been directly imported as warnings

### PR DESCRIPTION
**Pull request by jack870131**

### CODE REVIEW

**Review by Beyhum — APPROVED** [review]

File: server/src/validator.ts [hunk 1]
```diff
@@ -65,6 +65,23 @@ export function validateNamesAndMappings(hwDefinition: HardwareDefinition, inclu
...
+					const mappedToPath = reservedNames.get(mappedTo)?.hardwareDefinitionUri;
+					let hasIndirectImport = true;
+					for (const importedHwDefinition of hwDefinition.imports) {
+						if (mappedToPath == importedHwDefinition.uri) {
+							hasIndirectImport = false;
+						}
+					}
+					if (hasIndirectImport) {
+						const diagnostic: Diagnostic = {
+							message: `${mappedTo} is indirectly imported by ${hwDefinition.uri}`,
```
Comment (dorinbotan) [inline server/src/validator.ts:78]: Might be more useful to point to the file where this mapping has been imported from (mappedToPath), or, even better, the file where it must be included from (the one in hwDefinition.imports).

Example of option 1:
MT3620_GPIO06 indirectly imported from .../mt3620.json

Example of option 2:
MT3620_GPIO06 indirectly imported from .../mt3620_sk.json. Use ../avnet_mt3620_sk.json instead.

**Review by dorinbotan — CHANGES_REQUESTED** [review]
Minor change in the warning message can make it more informative.

File: server/src/validator.ts (hunk 1 above)
Comment (jack870131) [reply server/src/validator.ts]: Updated. Not sure if it follows what you suggested, so pls have a look again. Thx.

Comment (Beyhum) [reply server/src/validator.ts]: I think the amount of searching done to find the other pin mapping is a bit too expensive, keeping it as 
"MT3620_GPIO06 is indirectly imported from .../mt3620.json" might be better for now

**Review by dorinbotan — APPROVED** [review]
Looks good.

**Pull request merged**